crates/runestick/src/item.rs

```diff
@@ -346,17 +346,21 @@ impl Item {
 /// assert_eq!("hello::$0", Item::with_item(&[Str("hello"), Id(0)]).to_string());
 /// ```
 impl fmt::Display for Item {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use std::fmt::Write;
         let mut it = self.iter();
+        let mut f = "".to_owned();
 
         if let Some(last) = it.next_back() {
             for p in it {
                 write!(f, "{}::", p)?;
             }
 
-            write!(f, "{}", last)
+            write!(f, "{}", last)?;
+
+            formatter.pad(&f)
         } else {
-            write!(f, "{{root}}")
+            formatter.pad("{root}")
```

Comment:
This is ugly, but I found no better way - this makes sure padding specifiers in format strings work as intended.